### ISSUE TITLE
SMS won't come through on secondary profile, switch to admin profile and it is there.

### ISSUE THREAD

**Issue by beaudickey** [GrapheneOS/os_issue_tracker#3425]
I've been on grapheneos for two months now. I have an admin profile and two other profiles that I do everything on. When I get a new message on my secondary profile, I get the notification but then go to messages and nothing is there. I switch to admin profile, go to messages and it is there. I switch back to my secondary profile and then it shows up. I had zero issues until the lastest system update. Now I'm having the issue all day everyday. I've seen quite a few other people having the same issue throughout the years as well, but I did not see a solution.

**Comment by beaudickey** [comment]
I am using the default SMS app as well.

**Comment by adventure-tense** [comment]
I have a similar issue, except that it's the media that doesn't always get pushed to the secondary account.  I have to go back to the admin account to see what someone has texted me. Photo attachments seem to always show up on the admin account.  I noticed it the first time in this past May, and it's happens randomly, but frequently.  Have been thinking about installing my day-to-day account on the admin account.

Should I submit this as a separate bug?

**Comment by adventure-tense** [comment]
Since there doesn't seem to be much activity to resolve this since April, I'll go back to using my admin account and delete the secondary account. 

Not ideal, but practical.

**Comment by jfsanchez** [comment]
Same problem here. Of course calls and SMS activated on both profiles. At first (same months ago) I was able to get the SMS content once I switched to the main profile and switched back again. Now I am unable to receive any new SMS in the secondary profile even if I switch profiles. An old solution was having Google Messages in both profiles but that stopped working too. I use the same stock messages app right now and I've tested other SMS apps. Nothing works now.

**Comment by jbmagination** [comment]
I am also having similar issues. I do not receive texts as quickly as I need to on secondary profiles and only get them when I am on my Owner profile. This bug plus some other SMS-related bugs are making me stop using multi-user profiles. Granted, there are many known multi-user SMS bugs going on at the moment that can also be reproduced on stock.

**Comment by jfsanchez** [comment]
After several attempts, for me, the way to workaround it, is by using QLIK on the secondary profile and from settings selecting the option of Sync messages. I suppose QLIK tries to re-read the database again because it takes a long time but in the end, it shows the SMS. It doesn't matter which SMS app you use on the main profile: Stock, Google or QLIK.

I've done several tests with some SMS apps and the QLIK approach on secondary profile is the only one that works now with any SMS app in the main profile. I think the rest of the profiles are not being notified of new incoming SMS and by default when you load a new profile it doesn't check the messages database too. (Could it be reproduced with the old API in adb to simulate receiving SMS?). This is wishful thinking as I haven't read the logs since I'm no expert.

it would be great if someone could give us some ideas or advise on how to diagnose the situation. Is there a web with instructions that I'm not aware of?

**Comment by adventure-tense** [comment]
Is this being closed because its resolved? or for another reason?

**Comment by beaudickey** [comment]
> It should be resolved since [2025022700](https://grapheneos.org/releases#2025022700).

It definitely was not resolved. I ended up switching back to stock os due to this issue going on for 11 months.

**Comment by beaudickey** [comment]
No... I switched back to stock OS because this problem went on for 11 months on grapheneos with no fix.

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Mar 8, 2025 at 10:38 AM, Daniel ***@***.***> wrote:    
You're using SMS on secondary users in the stock OS? They don't have as many fixes for it...

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
thestinger left a comment (GrapheneOS/os-issue-tracker#3425)
You're using SMS on secondary users in the stock OS? They don't have as many fixes for it...

**Comment by beaudickey** [comment]
I did answer the question twice. You just don't know how to read and comprehend. The issue was happening when I was using grapheneos with a secondary profile. I have switched back to stock OS now because of the issue. 

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Mar 8, 2025 at 1:24 PM, Daniel ***@***.***> wrote:    
@beaudickey You didn't answer the question. Secondary user SMS has been much more broken on the stock OS than GrapheneOS.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
thestinger left a comment (GrapheneOS/os-issue-tracker#3425)
@beaudickey You didn't answer the question. Secondary user SMS has been much more broken on the stock OS than GrapheneOS.

**Comment by beaudickey** [comment]
Ok lol you clearly aren't understanding. I'm not using a secondary profile on stock OS. I was using it on grapheneos ONLY. I switched back to stock OS because of the many problems I had with grapheneos. I would not recommend using it.

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Mar 8, 2025 at 1:53 PM, Daniel ***@***.***> wrote:    
@beaudickey You have stated if you're using SMS on a secondary user in the stock OS. Secondary users aren't a GrapheneOS feature. SMS in a secondary user should be far more broken on the stock OS than GrapheneOS. It's odd to switch back to the stock OS because of something that works much better but still not perfectly on GrapheneOS.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
thestinger left a comment (GrapheneOS/os-issue-tracker#3425)
@beaudickey You have stated if you're using SMS on a secondary user in the stock OS. Secondary users aren't a GrapheneOS feature. SMS in a secondary user should be far more broken on the stock OS than GrapheneOS. It's odd to switch back to the stock OS because of something that works much better but still not perfectly on GrapheneOS.

**Comment by beaudickey** [comment]
I'm sure they are. That's why I'm not using it at all anymore

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, Mar 8, 2025 at 2:00 PM, Daniel ***@***.***> wrote:    
@beaudickey The secondary user SMS issues you're talking about are much worse on the stock Pixel OS than GrapheneOS.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
thestinger left a comment (GrapheneOS/os-issue-tracker#3425)
@beaudickey The secondary user SMS issues you're talking about are much worse on the stock Pixel OS than GrapheneOS.

**Comment by jfsanchez** [comment]
I've tried and it isn't solved for me (updated to 2025030300). When I receive a SMS in the main profile I can see the notification, not the content because it's a notification of another profile. The SMS is not received in the secondary profile on use.

Can I do something in order to help diagnose the problem?

**Comment by jfsanchez** [comment]
Google SMS (because of RCS) and QLIK View in the other profile to force the refresh. I can try with standard messaging app, however for me RCS is unfortunately a must.

**Comment by jfsanchez** [comment]
I've tried in other phone, a pixel 6A. I've uninstalled qlik and enabled the default SMS app in all profiles. It seems to work now. I'll do some tests with Google Messages and standard SMS app in my main phone and I'll post the results.

**Comment by jfsanchez** [comment]
I've tried in two phones: Pixel 8A and Pixel 6A. In both phones the standard messaging app works fine with profiles as long as this is the default SMS app on those profiles. Google Messages doesn't work even if it's enabled and as default app in both profiles.

I've also tried standard SMS app in the main profile and Google Messages in secondary profile, it didn't work, however I attempted this in only one phone.

It's a pity since for me RCS is a must for a couple of contacts. I wish Google could fix their app or GrapheneOS could implement RCS but being Google the main problem here, it seems unlikely.

**Comment by jfsanchez** [comment]
> [@jfsanchez](https://github.com/jfsanchez) Can you file a new issue about non-system SMS/MMS/RCS apps not working in secondary users? It's different than a generic SMS/MMS issue, which is resolved.

Thank you, done: https://github.com/GrapheneOS/os-issue-tracker/issues/4988
